tests/unit/modules/test_powerdns.py

```diff
@@ -0,0 +1,969 @@
+                    "type": "A",
+                    "ttl": 30,
+                }
+            ],
+        }
+
+        method_get_zone = mock.patch(
+            "salt.modules.powerdns._get_zone", return_value=old_zone
+        )
+
+        method_get_zone.start()
+        mock_patch_resp = self._mock_response(status=400, json_data="Test data")
+        mock_patch.return_value = mock_patch_resp
+
+        name = "zone1.com"
+        key = "TEST_KEY"
+        server = "pdns.test.com"
+        rrsets = [
+            {
+                "name": "zone1.com",
+                "records": [{"content": "12.34.56.78"}],
+                "type": "A",
+                "ttl": 30,
+            }
+        ]
+        nameservers = ["ns1.zone1.com", "ns2.zone2.com"]
+        drop_existing = False
+        dnssec = False
+
+        self.assertRaises(
+            CommandExecutionError,
+            powerdns.manage_zone,
+            name,
+            key,
+            server,
+            rrsets,
+            nameservers,
+            drop_existing,
+            dnssec,
+        )
+
+        method_get_zone.stop()
+
+    def test_delete_zone_no_zone(self):
+
+        name = "zone1.com"
+        key = "TEST_KEY"
+        server = "pdns.test.com"
+
+        method_get_zone = mock.patch(
+            "salt.modules.powerdns._get_zone", return_value=None
+        )
+        method_get_zone.start()
+
+        result = powerdns.delete_zone(name, key, server)
+
+        method_get_zone.stop()
+
+        self.assertEqual(result, {"changes": {}, "name": "zone1.com", "result": True})
+
+    @mock.patch("requests.delete")
+    def test_delete_zone_sucess(self, mock_delete):
+
+        name = "zone1.com"
+        key = "TEST_KEY"
+        server = "pdns.test.com"
+
+        mock_delete_resp = self._mock_response(status=204, json_data="Test data")
+        mock_delete.return_value = mock_delete_resp
+
+        method_get_zone = mock.patch(
+            "salt.modules.powerdns._get_zone", return_value={"name": "zone1.com"}
+        )
+        method_get_zone.start()
+
+        result = powerdns.delete_zone(name, key, server)
+
+        method_get_zone.stop()
+
+        self.assertEqual(
+            result,
+            {"changes": {"deleted": "zone1.com"}, "name": "zone1.com", "result": True},
+        )
+
+    @mock.patch("requests.delete")
+    def test_delete_zone_fails(self, mock_delete):
+
+        name = "zone1.com"
+        key = "TEST_KEY"
+        server = "pdns.test.com"
+
+        mock_delete_resp = self._mock_response(status=400, json_data="Test data")
+        mock_delete.return_value = mock_delete_resp
+
+        method_get_zone = mock.patch(
+            "salt.modules.powerdns._get_zone", return_value={"name": "zone1.com"}
+        )
+        method_get_zone.start()
+
+        self.assertRaises(
+            CommandExecutionError, powerdns.delete_zone, name, key, server
+        )
+
+        method_get_zone.stop()
```

Comment:
Since this is a new test module, would you mind rewriting it in `pytest` under `tests/pytests/unit/modules`?

Reply:
Rewrote.